CMakeLists.txt

```diff
@@ -6,8 +6,19 @@ if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
     set (CMAKE_INSTALL_PREFIX "${CMAKE_BINARY_DIR}/install" CACHE PATH "default install path" FORCE )
 endif()
 
+#-----------------------------------------------------------------------------
+# Build with Cuda Option - Enabled by default
+#-----------------------------------------------------------------------------
 option(BUILD_WITH_CUDA "Build With CUDA instead of OpenCL" ON)
```

Comment:
It usually better to prefix option with name of the project to ensure there a no conflict with existing option when vendoriz-ing.

`BUILD_WITH_CUDA` -> `Autoscoper_BUILD_WITH_CUDA`